### PR TITLE
feat: 🎸 change the deafult allocator to mimallocator

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#[global_allocator]
+static GLOBAL: mimallocator::Mimalloc = mimallocator::Mimalloc;
```
Comment (dsherret) [inline src/lib.rs:2]: This globally changes the allocator? Shouldn't people do this in their applications rather than this library doing it?

**Comment by IWANABETHATGUY** [comment]
According the flamegraph, our parser is a allocation heavy task , so change the default allocator to `mimallocator ` could increase 10%-15% performance, tested on 3mb json data. 
before: 62ms, after: 52ms;
https://github.com/swc-project/swc/blob/b9f5a50d184b2aa2c99631cea510d5762ecb1ea6/native/src/lib.rs#L6-L11

**Comment by IWANABETHATGUY** [comment]
the swc doing the same thing, they change the deafult allocator

**Comment by dsherret** [comment]
@IWANABETHATGUY thanks, but sorry, I believe this is a global change and so I wouldn't want to merge this. Applications may change their global allocator if they wish, but I don't believe this should be done by this crate.

Also, we should implement #9 before making any performance related improvements.

**Pull request closed without merging**